Source/SelfService/Web/microservice/dataTable.tsx

```diff
@@ -0,0 +1,211 @@
+
+    useEffect(() => {
+        const mapStatusToMicroservicesAndSetState = async () => {
+            const rows = await Promise.all(microservices.map(async microservice => {
+                const status = await getMicroserviceStatus(microservice.id);
+
+                return {
+                    ...microservice,
+                    status
+                } as MicroserviceObject;
+            }));
+
+            setRows(rows);
+        };
+
+        mapStatusToMicroservicesAndSetState();
+    }, [setRows]);
+
+    const getMicroserviceStatus = (microserviceId: string) => {
+        return new Promise((resolve, reject) => {
+            resolve(
+                getPodStatus(application.id, environment, microserviceId)
+                    .then(res => res.pods)
+                    .catch(err => reject(err))
+            );
+        });
+    };
+
+    const columns: GridColDef[] = [
+        {
+            field: 'name',
+            headerName: 'Name',
+            minWidth: 200,
+            flex: 1
+        },
+        {
+            field: 'image',
+            headerName: 'Container Image',
+            minWidth: 200,
+            flex: 1,
+            valueGetter: (params: GridValueGetterParams) =>
+                `${params.row.edit.extra.headImage || ''}`,
+        },
+        {
+            field: 'runtime',
+            headerName: 'Runtime',
+            minWidth: 200,
+            flex: 1,
+            valueGetter: (params: GridValueGetterParams) => {
+                const runtimeNumber = params.row.edit.extra.runtimeImage.replace(/[dolittle/runtime:]/gi, '');
+                return `${runtimeNumber || 'None'}`;
+            }
+        },
+        {
+            field: 'isPublic',
+            headerName: 'Public URL',
+            minWidth: 200,
+            flex: 1,
+            sortable: false,
+            // TODO: Tooltip needs public urls. Map them into title.
+            renderCell: function tooltip(params: GridRenderCellParams) {
+                return (
+                    <Tooltip title={``} arrow>
+                        <span>{params.row.edit.extra.isPublic ? 'Available' : 'None'}</span>
+                    </Tooltip>
+                );
+            },
+        },
+        {
+            field: 'status',
+            headerName: 'Status',
+            minWidth: 200,
+            flex: 1,
+            renderCell: function statusInfo(params: GridRenderCellParams) {
+                return <>{microserviceStatusInfo(params)}</>;
+            },
+            sortComparator: (v1, v2, param1: any, param2: any) => {
+                return param1.value[0].phase.localeCompare(param2.value[0].phase);
+            },
+        },
+    ];
+
+    const microserviceStatusInfo = (params: GridRenderCellParams) => {
+        const status = params.row.status[0].phase;
+        const checkStatus = status.toLowerCase();
+
+        let color = themeDark.palette.text.primary;
+        let icon = <ErrorRounded sx={{ color }} />;
```

Comment:
Would it maybe be better to have a question mark or something to indicate "unknown" as a default if we don't know what the status is?

Reply:
Yes, definitely.